### ISSUE TITLE
Not all data is saved to output file

### ISSUE THREAD

**Issue by noamha-ken** [taskrabbit/elasticsearch-dump#243]
ElasticDump version: 2.4.0
ElasitcSearch version: 2.3.3
Command:
elasticdump --input=http://192.168.70.106:9200/items/product --output=/tmp/product_data.json --type=data --limit 1000 --debug >> /tmp/product_direct.log

There are a few million items in the index.
At the beginning the output file begins to fill and grow to several MB very quickly, but then at some point it stop outputing a lot of records and just output a few singles records every few seconds.

The log continue to show a big number of records written:

```
Fri, 05 Aug 2016 04:09:21 GMT | got 1000 objects from source elasticsearch (offset: 0)
Fri, 05 Aug 2016 04:09:21 GMT | sent 1000 objects to destination file, wrote 1000
```

And the debug info I'm piping into a file is growing rapidly to a few hunderds MB and seems to continue indeed a big amount of records.
When looking into the debug info I see records that were not output to the data file.

In summary it seems like the records are retrieved but at some point only a friction of them saved to the data file

**Comment by evantahler** [comment]
Sadly, without a sample of your data which *isn't* dumped, there's nothing we can do to help you.  We've got tests for this project that don't exhibit this behavior.  

Is there any pattern to the data that isn't dumped out?  non UTF-8 characters?   Long data lengths?  Can you ensure that your disk isn't running out of disk and that your network connection hasn't been interrupted during the dump?

Good luck!

**Comment by noamha-ken** [comment]
There is no problem with the disk, like I said the "debug" information I'm piping to disk is working great and growing very fast.
The resulting size of the output was 46MB, while the debug output was 1,100MB which in deed included all the data.
Like I said it seems like the data is queried correctly but not written to the file.

I've tried another project "elasticsearch-tools" and it successfully output all the data to file.

I ran the same command against an index called "amazon_seller".

Here is a sample of the data itself from /tmp/product_data.json:
```
"_index":"amazon_seller","_type":"product","_id":"B00YSSVH6G","_score":1,"_source":{"asin":"B00YSSVH6G","name":"Popband Active Sport Hair Ties, Yoga Bunny Down Dog","brand":"Popband","rating":null,"price":9}}
{"_index":"amazon_seller","_type":"product","_id":"B014X7RSAU","_score":1,"_source":{"asin":"B014X7RSAU","name":"Popband Hair Ties, GI Jane","brand":"Popband","rating":4,"price":12}}
{"_index":"amazon_seller","_type":"product","_id":"B00WILZOYQ","_score":1,"_source":{"asin":"B00WILZOYQ","name":"Popband Hair Ties, Kasbah","brand":"Popband","rating":null,"price":12}}
{"_index":"amazon_seller","_type":"product","_id":"B00WMCF8S8","_score":1,"_source":{"asin":"B00WMCF8S8","name":"Popband Hair Ties, Monotone","brand":"Popband","rating":null,"price":12}}
{"_index":"amazon_seller","_type":"product","_id":"B00EAXRN5I","_score":1,"_source":{"asin":"B00EAXRN5I","name":"Pope Francis Framed Print House Blessing with Holy Card Free Cross Bookmark Included 10 by 12 Inches","brand":"E's Guardian Angel","rating":null,"price":39.99}}
{"_index":"amazon_seller","_type":"product","_id":"B0164DAIW6","_score":1,"_source":{"asin":"B0164DAIW6","name":"Pope Francis Ortodossa Cross Double Sided Medal Crucifix 3.5 Inches","brand":"Religious Faithful Gifts","rating":null,"price":15.99}}
{"_index":"amazon_seller","_type":"product","_id":"B019GKZOWI","_score":1,"_source":{"asin":"B019GKZOWI","name":"Popeye the Sailor Square Metal Watch","brand":"MNP","rating":null,"price":null}}
{"_index":"amazon_seller","_type":"product","_id":"B0119938G8","_score":1,"_source":{"asin":"B0119938G8","name":"Popfeel® Pack of 3 Applicator Round Sponge Powder Puff With Ribbon Ties Loose Powder Replacement...","brand":"Popfeel","rating":null,"price":5.59}}
{"_index":"amazon_seller","_type":"product","_id":"B0147PKF6C","_score":1,"_source":{"asin":"B0147PKF6C","name":"Popobe Bear Key Chain","brand":"","rating":null,"price":null}}
{"_index":"amazon_seller","_type":"product","_id":"B0168VYI98","_score":1,"_source":{"asin":"B0168VYI98","name":"Poppy Drops 3-Pack Temporary Tattoo \"Festive Favorites\" Earring Collections - Girls Rock, Glow in...","brand":"Poppy Drops","rating":null,"price":15.99}}
{"_index":"amazon_seller","_type":"product","_id":"B015XRKA5O","_score":1,"_source":{"asin":"B015XRKA5O","name":"Popular 24\" Dark Black Long Straight Womens Ponytail Clip in Pony Tail Hair Extensions Wrap on...","brand":"always_sexy","rating":3,"price":0.99}}
{"_index":"amazon_seller","_type":"product","_id":"B00KH5F40E","_score":1,"_source":{"asin":"B00KH5F40E","name":"Popular Mechanics Magazine [June 2014]","brand":"Popular Mechanics","rating":null,"price":9.99}}
{"_index":"amazon_seller","_type":"product","_id":"B00KQFH9LC","_score":1,"_source":{"asin":"B00KQFH9LC","name":"Popular Mechanics Magazine Subscription, Print, 1 Year, 10 Issues Per Year","brand":"","rating":null,"price":null}}
{"_index":"amazon_seller","_type":"product","_id":"B00SSENKOI","_score":1,"_source":{"asin":"B00SSENKOI","name":"Porcelain 2 Piece Christmas Snowman Sugar & Creamer Winter Wonders Set By Debbie Mumm's","brand":"Debbie Mumm","rating":null,"price":null}}
{"_index":"amazon_seller","_type":"product","_id":"B00NLO8O2S","_score":1,"_source":{"asin":"B00NLO8O2S","name":"Porcelain Teapot Teabag Caddy White Set of 4","brand":"R&M","rating":null,"price":9.99}}
{"_index":"amazon_seller","_type":"product","_id":"B00JE078KC","_score":1,"_source":{"asin":"B00JE078KC","name":"Pore Perfect Refine & Renew Gel Peel Off Mask By Kym's Diva Stuff,2oz","brand":"Diva Stuff","rating":3.6,"price":11.99}}
{"_index":"amazon_seller","_type":"product","_id":"B00S9JQ7OM","_score":1,"_source":{"asin":"B00S9JQ7OM","name":"Poreless Perfection Foundation, Medium 0.39 oz (11 g)","brand":"Mally","rating":5,"price":47.95}}
{"_index":"amazon_seller","_type":"product","_id":"B00SP9IZY6","_score":1,"_source":{"asin":"B00SP9IZY6","name":"Porfirio Cadena El Ojo De Vidrio En \"Vuelve Porfirio Cadena\" Radionovela Mp3","brand":"N/A","rating":null,"price":null}}
```

And here is a sample of the debug output:
```
Fri, 05 Aug 2016 04:22:31 GMT | starting dump
Fri, 05 Aug 2016 04:22:31 GMT | got 1000 objects from source elasticsearch (offset: 0)
Fri, 05 Aug 2016 04:22:31 GMT | sent 1000 objects to destination file, wrote 1000
Fri, 05 Aug 2016 04:22:31 GMT [debug] | scrollRequest: {"uri":"http://192.168.70.106:9200/_search/scroll?scroll=10m","method":"GET","body":"cXVlcnlUaGVuRmV0Y2g7NTs4NTpZclRnWFFHM1NmV29Oei03LTRqSkZ3Ozg3OllyVGdYUUczU2ZXb056LTctNGpKRnc7ODY6WXJUZ1hRRzNTZldvTnotNy00akpGdzs4ODpZclRnWFFHM1NmV29Oei03LTRqSkZ3Ozg5OllyVGdYUUczU2ZXb056LTctNGpKRnc7MDs="}
Fri, 05 Aug 2016 04:22:31 GMT [debug] | body: "{\"_scroll_id\":\"cXVlcnlUaGVuRmV0Y2g7NTs4NTpZclRnWFFHM1NmV29Oei03LTRqSkZ3Ozg3OllyVGdYUUczU2ZXb056LTctNGpKRnc7ODY6WXJUZ1hRRzNTZldvTnotNy00akpGdzs4ODpZclRnWFFHM1NmV29Oei03LTRqSkZ3Ozg5OllyVGdYUUczU2ZXb056LTctNGpKRnc7MDs=\",\"took\":22,\"timed_out\":false,\"terminated_early\":true,\"_shards\":{\"total\":5,\"successful\":5,\"failed\":0},\"hits\":{\"total\":4387119,\"max_score\":1.0,\"hits\":[{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B00BKTPYWE\",\"_score\":1.0,\"_source\":{\"asin\":\"B00BKTPYWE\",\"name\":\"Kelo-cote Advanced Formula Scar Gel 0.71 oz\",\"brand\":\"AB\",\"rating\":null,\"price\":null}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B003OFAT6I\",\"_score\":1.0,\"_source\":{\"asin\":\"B003OFAT6I\",\"name\":\"Kelo-cote Advanced Formula Scar Gel 60g\",\"brand\":\"Kelo-Cote\",\"rating\":3.0,\"price\":105.27}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B00ZK60GZI\",\"_score\":1.0,\"_source\":{\"asin\":\"B00ZK60GZI\",\"name\":\"Kelo-cote Advanced Formula Silicone Scar Gel, 15 Gram Tube\",\"brand\":\"Kelo-Cote\",\"rating\":3.7,\"price\":69.99}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B00P84YJXW\",\"_score\":1.0,\"_source\":{\"asin\":\"B00P84YJXW\",\"name\":\"Ken Orrick and his Uncle J.T Young - Sing Gospel Favorites [Audio CD]\",\"brand\":\"Ken Orrick\",\"rating\":null,\"price\":null}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B011MFFFTM\",\"_score\":1.0,\"_source\":{\"asin\":\"B011MFFFTM\",\"name\":\"Kenchii Five Star Offset 46-tooth 6.5\\\" Thinning Shears\",\"brand\":\"The Sharper Edge\",\"rating\":null,\"price\":124.2}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B00J7LWZHY\",\"_score\":1.0,\"_source\":{\"asin\":\"B00J7LWZHY\",\"name\":\"K-Concept Chrome Plated Barber Pole, Bp1238d\",\"brand\":\"K-Concept\",\"rating\":5.0,\"price\":101.44}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B004WP4U5A\",\"_score\":1.0,\"_source\":{\"asin\":\"B004WP4U5A\",\"name\":\"K-pak Color Therapy Shampoo ( to Preserve Color & Repair Damaged ) - 1000ml/33.8oz\",\"brand\":\"Joico\",\"rating\":4.4,\"price\":26.99}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B008MAN4NA\",\"_score\":1.0,\"_source\":{\"asin\":\"B008MAN4NA\",\"name\":\"K-Pop Kim Hyung Jun - Escape [Package 1 : CD + Art Book 60P] (KHJ02A)\",\"brand\":\"\",\"rating\":4.5,\"price\":24.99}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B017LIJWAW\",\"_score\":1.0,\"_source\":{\"asin\":\"B017LIJWAW\",\"name\":\"K.O.P KING OF POP :: Hisashi Eguchi All Illustration Collection Works ???? ??????????? [ART BOOK...\",\"brand\":\"\",\"rating\":null,\"price\":null}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B0108R0G3K\",\"_score\":1.0,\"_source\":{\"asin\":\"B0108R0G3K\",\"name\":\"K0580 (M1141B) Alternative Repair Kits for 5D2 Solenoid Valves\",\"brand\":\"\",\"rating\":null,\"price\":null}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B00G2U3O50\",\"_score\":1.0,\"_source\":{\"asin\":\"B00G2U3O50\",\"name\":\"K1F NFL DETROIT LIONS Refillable Butane Torch Lighter - Tin Gift Box - 2 1/4 Inch - Factory New -\",\"brand\":\"NFL\",\"rating\":null,\"price\":6.5}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B005E9YFA8\",\"_score\":1.0,\"_source\":{\"asin\":\"B005E9YFA8\",\"name\":\"K9.5 \\\"We Are the Dogs\\\" - Reading & Language Arts Adventures 2 - K9.5 Series - Lightspan...\",\"brand\":\"Lightspan Adventures\",\"rating\":null,\"price\":null}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B00A91OA74\",\"_score\":1.0,\"_source\":{\"asin\":\"B00A91OA74\",\"name\":\"KA'OIR By Keyshia KA'OIR \\\"Choco Latte\\\" Chocolate brown Lipstick KAOIR\",\"brand\":\"KA'OIR Cosmetics\",\"rating\":4.3,\"price\":18.99}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B005G3X9P4\",\"_score\":1.0,\"_source\":{\"asin\":\"B005G3X9P4\",\"name\":\"KA'OIR By Keyshia KAOIR \\\"Baby Blue\\\" Lipstick\",\"brand\":\"KA'OIR Cosmetics\",\"rating\":4.0,\"price\":17.99}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B00KEECS3O\",\"_score\":1.0,\"_source\":{\"asin\":\"B00KEECS3O\",\"name\":\"KA'OIR By Keyshia KAOIR \\\"Banana Milkshake\\\" Bright Yellow Lipstick only\",\"brand\":\"KA'OIR Cosmetics\",\"rating\":4.5,\"price\":18.99}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B00HBF31US\",\"_score\":1.0,\"_source\":{\"asin\":\"B00HBF31US\",\"name\":\"KA'OIR By Keyshia KAOIR \\\"BOSSED UP\\\" Wet Paint Lip Gloss NUDE\",\"brand\":\"KA'OIR\",\"rating\":null,\"price\":19.99}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B00927I68K\",\"_score\":1.0,\"_source\":{\"asin\":\"B00927I68K\",\"name\":\"KA'OIR By Keyshia KAOIR \\\"GOLDEN GODDESS\\\" GOLD Lipstick GLITZSTICK Glitter Set\",\"brand\":\"KA'OIR Cosmetics\",\"rating\":5.0,\"price\":27.99}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B00927JJTK\",\"_score\":1.0,\"_source\":{\"asin\":\"B00927JJTK\",\"name\":\"KA'OIR By Keyshia KAOIR \\\"MISTRESS\\\" Hot PINK Lipstick GLITZSTICK Glitter Set\",\"brand\":\"KA'OIR Cosmetics\",\"rating\":5.0,\"price\":27.99}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B005G4959M\",\"_score\":1.0,\"_source\":{\"asin\":\"B005G4959M\",\"name\":\"KA'OIR By Keyshia KAOIR \\\"Nudity\\\" Nude Lipstick\",\"brand\":\"KA'OIR Cosmetics\",\"rating\":2.7,\"price\":16.99}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B00640Y12S\",\"_score\":1.0,\"_source\":{\"asin\":\"B00640Y12S\",\"name\":\"KA'OIR By Keyshia KAOIR \\\"Pool Party\\\" Turquoise Bright Blue Lipstick\",\"brand\":\"KA'OIR Cosmetics\",\"rating\":4.6,\"price\":19.99}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B00HBF66UK\",\"_score\":1.0,\"_source\":{\"asin\":\"B00HBF66UK\",\"name\":\"KA'OIR By Keyshia KAOIR \\\"SUPERSTAR\\\" Wet Paint Lip Gloss MINT GREEN\",\"brand\":\"KA'OIR\",\"rating\":4.0,\"price\":19.99}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B009TNGANA\",\"_score\":1.0,\"_source\":{\"asin\":\"B009TNGANA\",\"name\":\"Kaaral Pink Up Hair Fix Extra Strong Spray - 12 oz\",\"brand\":\"kaaral\",\"rating\":5.0,\"price\":10.99}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B009TNGAG2\",\"_score\":1.0,\"_source\":{\"asin\":\"B009TNGAG2\",\"name\":\"Kaaral Pink Up Straight Up Root Lifter - 5.25 oz\",\"brand\":\"kaaral\",\"rating\":null,\"price\":21.49}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B00BFW9KKS\",\"_score\":1.0,\"_source\":{\"asin\":\"B00BFW9KKS\",\"name\":\"Kaaral Style Perfetto Fortissimo Extra Hold Gel - 8.8 oz\",\"brand\":\"kaaral\",\"rating\":5.0,\"price\":13.75}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B019TAEA52\",\"_score\":1.0,\"_source\":{\"asin\":\"B019TAEA52\",\"name\":\"Kadi Natural Braiding Style, Hair Extension, Professional Grade Kanekalon (T1B/350 - Black...\",\"brand\":\"Kadi Natural\",\"rating\":4.0,\"price\":12.5}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B012XO5XNY\",\"_score\":1.0,\"_source\":{\"asin\":\"B012XO5XNY\",\"name\":\"Kadiya Black Mix Blue Short Men Hair Synthetic Cosplay Anime Wig Kageyama Tobio Haikyuu!!\",\"brand\":\"KADIYA\",\"rating\":5.0,\"price\":14.99}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B00VFCJLN4\",\"_score\":1.0,\"_source\":{\"asin\":\"B00VFCJLN4\",\"name\":\"Kadiya GinTama DEATH NOTE-Sakata Gintoki Near Halloween Cosplay wig\",\"brand\":\"KADIYA\",\"rating\":5.0,\"price\":12.99}},{\"_index\":\"amazon_seller\",\"_type\":\"product\",\"_id\":\"B00MGWTZI8\",\"_score\":1.0,\"_source\":{\"asin\":\"B00MGWTZI8\",\"name\":\"Kadiya One Piece Nami Medium Short Orange Halloween Anime Cosplay Wig Free Hair Cap\",\"brand\":\"KADIYA\",\"rating\":null,\"price\":12.99}}
```

Thanks

**Comment by noamha-ken** [comment]
I did some more testing and tried to debug it.
When copying the same data from URL to URL it works perfectly and synced all the data.
I've also tried adding some logging output, and again it seems like it's reading all the data without any errors, the problem is with writing it to the file which for some reason doesn't really output to the text file.
My guess it's something with output object which doesn't really save all the records.

There are around 4.3 million records.

Thanks,
Noam.

**Comment by evantahler** [comment]
Can you clarify: The log file still says `sent 1000 objects to destination file, wrote 1000`, meaning it *thinks* that it is always writing all the documents it got?

Which OS and node.js version are you using?

**Comment by Nagesh74** [comment]
Hi,
 I am seeing lot of discussion happening here about download logs from Kibana 4. where I can find the link to logs download in Kibana 4. Please help me.

**Comment by raffi-minassian** [comment]
If some **documents** contain a field named **_id** , I also see this problem .The write fails silently and you will get the result you are seeing. Not sure why this occurs. I think **_id** is a reserved field. It can not be in your _document_. Not sure how that messes up the writing to file exactly.

**Comment by mysticalmuse11** [comment]
It sounds like you're running into an issue where not all data is being saved to your output file. That can definitely be frustrating! It might help to check if there are any chunker settings that need adjusting. Also, if you're working with [thinning shears](https://magixshears.com/dog-grooming-chunker-shears/) for your project, adding a "chunker" feature could help refine the output even more, similar to how thinning shears give a clean, efficient cut. If you need any tips or troubleshooting help with that, feel free to reach out!
